### DISCUSSION TITLE
Video

### DISCUSSION THREAD

**Discussion by mschingel** [tsightler/ring-mqtt#406] [Support Q&A]
Is there a video on how to get the video working?  I know you said we have the setup the camera and read the documentation.  But I don't understand where to begin...  I been using the "Ring" Integration vs this one, however the delay stinks.

**Comment by tsightler** [comment]
The ring-mqtt project simply exposes the Ring cameras as generic RTSP cameras.  Setting these cameras up is a Home Assistant function, so it's covered in the docs for HA, but there are probably videos somewhere, but overall the process is quite simple.

https://www.home-assistant.io/integrations/generic/

If you have specific issues or questions with the process please feel free to ask and I'll try help.

**Comment by mschingel** [comment]
Figured it out.  It was the image address. In the log it’s https. But
really it’s http.

**Comment by mschingel** [comment]
Thanks,

I just tried and it forces me to select RSTP transport protocol. I selected
TCP

I entered the values I found in the ring-Mqtt Log which are the same
outlined in your steps in the read me.

I get the “unknown error occurred”

**Comment by Rincovanderbeek** [comment]
Good morning,
I have setup your nice MQTT ring application and most of it works fine.
Only I don't get the live stream to work
I use the rtsp stream mentioned in the info attributes, it cannot be opened.

The stream looks like this (expect the xx):
rtsp://xxxxxxxxx-ring-mqtt:8554/xxxxxxda18_live

Any suggestions?

**Reply by eggtastico** [reply]
rtsp://xxxxxxxxx-ring-mqtt:8554/xxxxxxda18_live
Lets break this down :-
rstp://<Host_Name>:<PORT>/<YOUR_MAC_ADDRESS>_live

The 03cabcc9-ring-mqtt does not change. Or you can confirm this by going to settings -> addons in your Home Assistant
Open Ring-MQTT with Video Streaming Addon & check the Hostname
Hostname
03cabcc9-ring-mqtt

You can get the mac address from within the Ring App on  your mobile device
Open Ring, goto Devices, Device Health, scroll to MAC address. Ignore the : so AB:CD:EF:12:34:56 would simply be ABCDEF123456

So you should end up wiht rsp of something like :- 
 rtsp://03cabcc9-ring-mqtt:8554/ABCDEF123456_live
 rtsp:// ----Hostname------:PORT/--MacAddress--_live

Username & Password required is your Ring Account details
Make sure the entites matches the name of your home assistant entity that starts with camera & ends in _snapshot. This usually matches the name of the device in the Ring App (so front_door, front_porch, garage), etc.
camera.<ENTITY_NAME>_snapshot
